traitsui/tests/editors/test_liststr_editor_selection.py

```diff
@@ -74,21 +92,539 @@ class ListStrEditorWithSelectedIndex(HasTraits):
 
-    selected = []
-    item = -1
-    while True:
-        item = control.GetNextItem(
-            item, wx.LIST_NEXT_ALL, wx.LIST_STATE_SELECTED
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def clear_selection(editor):
+    if is_current_backend_wx():
+        import wx
+
+        currently_selected = get_selected_indices(editor)
+        # Deselect all currently selected items
+        for selected_index in currently_selected:
+            editor.control.SetItemState(
+                selected_index, 0, wx.LIST_STATE_SELECTED
+            )
+
+    elif is_current_backend_qt4():
+        editor.list_view.selectionModel().clearSelection()
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+def right_click_item(control, index):
+
+    if is_current_backend_wx():
+        import wx
+
+        event = wx.ListEvent(
+            wx.EVT_LIST_ITEM_RIGHT_CLICK.typeId, control.GetId()
         )
-        if item == -1:
-            break
-        selected.append(item)
-    return selected
+        event.SetIndex(index)
+        wx.PostEvent(control, event)
+
+    elif is_current_backend_qt4():
+        # Couldn't figure out how to close the context menu programatically
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+    else:
+        raise unittest.SkipTest("Test not implemented for this toolkit")
+
+
+@unittest.skipIf(is_current_backend_wx(), "Issue enthought/traitsui#752")
+@skip_if_null
+class TestListStrEditor(unittest.TestCase):
+
+    def setup_gui(self, model, view):
+        gui = GUI()
+        ui = model.edit_traits(view=view)
+        self.addCleanup(ui.dispose)
+
+        gui.process_events()
+        editor = ui.get_editors("value")[0]
+
+        return gui, editor
+
+    def test_list_str_editor_single_selection(self):
+        with store_exceptions_on_all_threads():
+            gui, editor = self.setup_gui(ListStrModel(), get_view())
+
+            if is_current_backend_qt4():  # No initial selection
+                self.assertEqual(editor.selected_index, -1)
+                self.assertEqual(editor.selected, None)
+            elif is_current_backend_wx():  # First element selected initially
+                self.assertEqual(editor.selected_index, 0)
+                self.assertEqual(editor.selected, "one")
+
+            set_selected_single(editor, 1)
+            gui.process_events()
+
+            self.assertEqual(editor.selected_index, 1)
+            self.assertEqual(editor.selected, "two")
+
+            set_selected_single(editor, 2)
+            gui.process_events()
+
+            self.assertEqual(editor.selected_index, 2)
+            self.assertEqual(editor.selected, "three")
+
+            clear_selection(editor)
+            gui.process_events()
+
+            self.assertEqual(editor.selected_index, -1)
+            self.assertEqual(editor.selected, None)
```

Comment:
I didn't realise that those attributes are internal state. Since factory description of `selected` and `selected_index` says "optional", I understood it as "can be used to observe selection if convenient". And in this case observing on the editor was more convenient.
Can change the tests to observe via the model instead, but the tests might be more fragile because then all of them would also be testing the behaviour of `sync_value`.

Reply:
I'd be in favour of using the model state if that is possible, assuming those editor states are internal states. The tests should not be fragile if `sync_value` is correct. (However if `sync_value` is incorrect, we should open a separate issue for that.)

Reply:
I started to make the changes, but I've run into quite a few issues where values aren't synced as expected. Given that the tests are for this specific editor, I would prefer not to start digging into the syncing code and leave the tests as they are.

Reply:
Sigh I had worried that might happen and was hoping it *might* be okay... Thank you for trying though. Agreed it is not worth diving into this :hole: for this.

Reply:
If you notice anything from `sync_value` that is worth a separate issue, please open them but feel free to leave them if it should require more investigations.